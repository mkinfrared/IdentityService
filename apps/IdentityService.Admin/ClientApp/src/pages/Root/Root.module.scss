.Root {
  display: flex;
  flex: 1;
}
