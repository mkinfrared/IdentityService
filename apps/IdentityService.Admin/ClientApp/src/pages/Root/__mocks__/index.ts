export { default } from "./Root.mock";
