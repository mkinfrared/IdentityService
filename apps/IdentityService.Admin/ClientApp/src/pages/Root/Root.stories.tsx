import { Meta, StoryObj } from "@storybook/react";

import Root from "./Root";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Root",
  component: Root,
  tags: ["autodocs"],
} satisfies Meta<typeof Root>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Root {...args} />
    </div>
  ),
};

export { Default };

export default meta;
