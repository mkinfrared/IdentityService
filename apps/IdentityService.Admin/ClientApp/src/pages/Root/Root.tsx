import { classNames } from "@identity-service/core";
import { Outlet } from "react-router-dom";

import { useUserSession } from "shared/lib/hooks";
import { Main } from "widgets/main";
import { MainSidebar } from "widgets/mainSidebar";

import css from "./Root.module.scss";
import { RootProps } from "./Root.type";

const Root = ({ className }: RootProps) => {
  const user = useUserSession();

  return (
    <div className={classNames(css.Root, className)} data-testid="Root">
      {user && <MainSidebar className={css.sidebar} />}
      <Main className={css.main}>
        <Outlet />
      </Main>
    </div>
  );
};

export default Root;
