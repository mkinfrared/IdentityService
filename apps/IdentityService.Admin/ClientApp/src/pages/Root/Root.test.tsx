import { render } from "@testing-library/react";

import Root from "./Root";

describe("<Root />", () => {
  const Component = <Root />;

  it("should be defined", () => {
    expect(Root).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Root");

    expect(element).toBeDefined();
  });
});
