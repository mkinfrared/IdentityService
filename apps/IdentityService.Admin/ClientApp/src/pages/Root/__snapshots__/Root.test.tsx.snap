// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should match the snapshot 1`] = `
<div>
  <div
    class="Root"
    data-testid="Root"
  >
    <main
      class="Main main"
      data-testid="Main"
    >
      <section
        class="page"
      />
    </main>
  </div>
</div>
`;
